src/compositor/lipstickcompositor.cpp

```diff
@@ -622,20 +624,38 @@ void LipstickCompositor::setScreenOrientation(Qt::ScreenOrientation screenOrient
     }
 }
 
-QString LipstickCompositor::keyboardLayout() const
+LipstickKeymap *LipstickCompositor::keymap() const
 {
-    return m_keyboardLayout;
+    return m_keymap;
 }
 
-void LipstickCompositor::setKeyboardLayout(const QString &layout)
+void LipstickCompositor::setKeymap(LipstickKeymap *keymap)
 {
-    if (layout != m_keyboardLayout) {
-        m_keyboardLayout = layout;
-        defaultInputDevice()->setKeymap(QWaylandKeymap(m_keyboardLayout));
-        emit keyboardLayoutChanged();
+    if (!keymap) {
+        qWarning() << "Cannot set an empty keymap";
+        return;
+    }
+
+    if (!m_keymap || (*m_keymap != *keymap)) {
```

Comment:
The "(*m_keymap != *keymap)" is odd, imho. This method should set *that* keymap object, and return early only if we're setting the same object as the current one.